pkg/controllers/nodeclass/status/status_condition.go

```diff
@@ -0,0 +1,50 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
+)
+
+type NodeClassStatus struct{}
+
+func (n NodeClassStatus) Reconcile(_ context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
+	if len(nodeClass.Status.AMIs) == 0 {
+		return reconcile.Result{}, fmt.Errorf("unable to resolve AMIs for node class")
+	}
+	if len(nodeClass.Status.Subnets) == 0 {
+		return reconcile.Result{}, fmt.Errorf("unable to resolve subnets for node class")
+	}
+	if len(nodeClass.Status.SecurityGroups) == 0 {
+		return reconcile.Result{}, fmt.Errorf("unable to resolve security groups for node class")
+	}
+	if len(nodeClass.Status.InstanceProfile) == 0 {
+		return reconcile.Result{}, fmt.Errorf("unable to resolve instance profile for node class")
+	}
+	readyCondition := metav1.Condition{
+		Type:               v1beta1.NodeClassReady,
+		Status:             metav1.ConditionTrue,
+		LastTransitionTime: metav1.Time{},
+		Reason:             "DependentsResolved",
```

Comment:
What does kubelet do when it goes ready with its dependencies?

Reply:
I can't remember if we provide reasons on status conditions when the status condition is true, can we validate?

Reply:
I think we do. Verified this for `node` object
`Ready            True    Sun, 05 May 2024 20:16:20 -0700   Sun, 05 May 2024 20:16:20 -0700   KubeletReady                 kubelet is posting ready status`

I have removed it for now. Let me know if you think that we should add it back?

Reply:
Yeah, I think that we don't do this right now with our other status conditions, but I know upstream does do it here and there